support-models/src/main/scala/com/gu/support/workers/states/FailureHandlerState.scala

```diff
@@ -12,11 +12,19 @@ case class FailureHandlerState(
   user: User,
   giftRecipient: Option[GiftRecipient],
   product: ProductType,
-  paymentFields: Option[Either[PaymentFields, RedemptionData]], //Will be present if CreatePaymentMethod failed
-  paymentMethod: Option[Either[PaymentMethod, RedemptionData]], //Will be present if anything after CreatePaymentMethod failed
```

Comment:
FailureHandler needs the PaymentProvider to log to the ELK stack.
It extracted it from the paymentFields/paymentMethod, and passed it on (as an option) on to the ELK stack. 
 paymentprovider is not actually optional, it's just due to how the types worked out and the way it grabbed it.  

With the new PaymentMethodWithSchedule type coming in the email state in https://github.com/guardian/support-frontend/pull/2572/files#r445743286 - I could have would have added a third Option in the FailureState to continue the current approach.

Instead I've just unified it into a single simple non-optional field.